src/main/java/com/friendly_machines/intellij/plugins/ideanative2debugger/impl/DebugProcess.java

```diff
@@ -257,65 +265,55 @@ private SuspendContext generateSuspendContext(List<Object> threads, String curre
         return new SuspendContext(this, stacks.toArray(new ExecutionStack[0]), activeStackId);
     }
 
-    public List<Map<String, ?>> getVariables(String threadId, String frameId) throws GdbMiOperationException, ClassCastException, IOException, InterruptedException {
+    public List<Map<String, ?>> getVariables(String threadId, String frameId) throws GdbMiOperationException, IOException, InterruptedException {
         // TODO: --simple-values and find stuff yourself.
-        var q = gdbCall("-stack-list-variables", List.of("--thread", threadId, "--frame", frameId, "--all-values"), Collections.emptyList());
-        try {
-            var variables = (Collection<Map<String, ?>>) q.get("variables");
-            if (variables == null) {
-                return Collections.emptyList();
-            }
-            // pucgenie: Why even copy?
-            return new ArrayList<Map<String, ?>>(variables);
-        } catch (ClassCastException e) {
-            e.printStackTrace();
-            throw new RuntimeException(e);
+        var q = gdbCall("-stack-list-variables", List.of("--thread", threadId, "--frame", frameId), List.of("all-values"));
```

Comment:
I think that won't work because it encodes into `--thread 1 --frame 2 --all-values -- all-values`, but we would need `--thread 1 --frame 2 --all-values -- all-values`.

The `--all-values` is an option, but the `--` on its own is just the end-of-options mark.